### PR TITLE
New short section on AI/ML

### PR DESCRIPTION
**Pull request by david-a-wheeler**
People will use AI & ML whether or not we say anything, so let's say something. There's a lot we can say, here is what I think are the basics.

Unfortunately every evasion countermeasure I've seen analyzed has been shown ineffective. That's hard to word, but I think a warning is needed. Here's my chop:

> As of this writing we don't know of any countermeasure for evasion
> (adversarial inputs) with a strong industry-wide
> confidence that it fully counters
> determined adversaries (we would love to know of one).

I looked years ago & came up empty. But that may be old news. I'd love to get an update.

Signed-off-by: David A. Wheeler <dwheeler@dwheeler.com>

### CODE REVIEW

**Comment by david-a-wheeler** [comment]
Note: Click on the "Files changed" tab to see the specific proposed text.

**Comment by david-a-wheeler** [comment]
I'm told there's a typo in this PR. I'm awaiting a tweaked PR or comment specifics to tell me where, so I can fix it and then merge this change. My intent is to merge this soon (once the typo is fixed).

**Review by lehors — APPROVED** [review]
LGTM. A few missing words but I think this is the right level of information for this material. Good job David.

File: secure_software_development_fundamentals.md [hunk 1]
```diff
@@ -4976,6 +4976,109 @@ When disposing, make sure you fully destroy any data you are supposed to destroy
...
+used by its users.
+Many ML projects have assumed a closed and trusted environment where
+there are no security threats.
+However, this assumption is often unrealistic.
+
+*Adversarial machine learning* is the set of efforts to
+protect the ML pipeline to ensure its security during training,
+test, and inference.
+This is an active area of study, and terminology varies.
+That said, there many kinds of potential attacks on ML systems, including:
```
[the remark this answers is not in the document]
Comment (david-a-wheeler) [reply secure_software_development_fundamentals.md:5002]: Agreed!

File: secure_software_development_fundamentals.md [hunk 2]
```diff
@@ -4976,6 +4976,109 @@ When disposing, make sure you fully destroy any data you are supposed to destroy
...
+  In a poisoning attack, the attacker manipulate data that will be used as
+  training data, e.g., to reduce performance, cause misclassification, and/or
+  insert backdoors
+  (Nicolae et al, 2019).
+  ML systems typically need a large amount of training data;
+  some attackers may even create or manipulate publicly-available
+  data if it is likely to be eventually used for training.
+  This kind of attack may lead to a loss of integrity and/or availability.
+* *Loss of confidentiality* ("reveal the wrong thing").
+  An attacker may be able use query results to reveal hidden information.
```
[the remark this answers is not in the document]
Comment (david-a-wheeler) [reply secure_software_development_fundamentals.md:5025]: Agreed!

File: secure_software_development_fundamentals.md [hunk 3]
```diff
@@ -4976,6 +4976,109 @@ When disposing, make sure you fully destroy any data you are supposed to destroy
...
+determine if the countermeasures will be adequate for your purposes.
+Many countermeasures only work against naive attackers who do not
+compensate for countermeasures.
+Depending on your purposes,
+there may not be *any* countermeasure that adequately counters attackers
+with adequate confidence.
+*Many* countermeasures have been proposed and later found to be inadequate.
+One paper that discusses how to evaluate countermeasures is by
+[Nicholas Carlini, Anish Athlye, Nicolas Papernot, et al., “On Evaluating Adversarial Robustness”, 2019-02-20](https://arxiv.org/pdf/1902.06705).
+We hope that in future there will be better countermeasures with
```
[the remark this answers is not in the document]
Comment (david-a-wheeler) [reply secure_software_development_fundamentals.md:5079]: Agreed!

**Pull request merged**